app/controllers/dashboards/base_controller.rb

```diff
@@ -58,5 +60,30 @@ def entered
       render layout: !request.xhr?
     end
 
+    def months
+      months = {}
+      active_report_class.ordered.select(:id, :parameters).index_by(&:parameters).each do | key, report |
+        report.set_date_range
+        start_date = report.range.start
+        months[report.id] = "#{Date::MONTHNAMES[start_date.month]} #{start_date.year}"
+      end
+      months
+    end
+
+    def selected_report_id
+      params[:choose_report][:month].to_i rescue id_of_most_recent_report
+    end
+
+    def id_of_most_recent_report
+      # assumes that newer reports have higher ids -- may not hold on dev
+      months.max_by {|k, v| k}[0] # id is key of the map
+    end
+
+    def selected_report_for (report_class)
+       selected_report = active_report_class.find(selected_report_id)
+      report_class.where(
+          created_at: [selected_report.created_at.beginning_of_day..selected_report.created_at.end_of_day])
+          .limit(1).first
```

Comment:
Another stylistic request.  The `.` should come at the end of the line instead of the beginning to indicate the previous statement isn't complete.  (Mostly this helps with copy and paste into a console since leaving it off will execute immediately.)